### PR TITLE
[AVT] - Added keyboard nav test `Breadcrumb`

### LINKED ISSUE #14638: Add Keyboard Nav AVT test for `Breadcrumb`

### PR DESCRIPTION
**Pull request by guidari**
Closes #14638 

Added keyboard navigation to `Breadcrumb` component.

#### Changelog

**New**

- Added new file `e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js`

#### Testing / Reviewing

- Ensure the test pass.
- Ensure that all the necessary navigation is covered.

### CODE REVIEW

File: e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+import { expect, test } from '@playwright/test';
+import { visitStory } from '../../test-utils/storybook';
+
+test.describe('breadcrumb @avt', () => {
+  test('accessibility-checker', async ({ page }) => {
```
Comment (tay1orjones) [inline e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js]:
```suggestion
  test('@avt-default-state', async ({ page }) => {
```

File: e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      component: 'Breadcrumb',
+      id: 'components-breadcrumb--default',
+      globals: {
+        theme: 'white',
+      },
+    });
+    await expect(page).toHaveNoACViolations('Breadcrumb');
+  });
+
+  test('accessibility-checker with overflow menu', async ({ page }) => {
```
Comment (tay1orjones) [inline e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js]:
```suggestion
  test('@avt-advanced-states with overflow menu', async ({ page }) => {
```

File: e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      component: 'Breadcrumb',
+      id: 'components-breadcrumb--breadcrumb-with-overflow-menu',
+      globals: {
+        theme: 'white',
+      },
+    });
+    await expect(page).toHaveNoACViolations('Breadcrumb-with-overflow-menu');
+  });
+
+  test('accessibility-checker skeleton', async ({ page }) => {
```
Comment (tay1orjones) [inline e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js]:
```suggestion
  test('@avt-advanced-states skeleton', async ({ page }) => {
```

File: e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      component: 'Breadcrumb',
+      id: 'components-breadcrumb--skeleton',
+      globals: {
+        theme: 'white',
+      },
+    });
+    await expect(page).toHaveNoACViolations('Breadcrumb-skeleton');
+  });
+
+  test('accessibility-checker keyboard navigation', async ({ page }) => {
```
Comment (tay1orjones) [inline e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js]:
```suggestion
  test('@avt-keyboard-nav', async ({ page }) => {
```

File: e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    await expect(page.getByText('Breadcrumb 1')).toBeFocused();
+
+    // Testing link. For this test pass the href must be like that: href="#"
+    await page.keyboard.press('Enter');
+    await expect(page).toHaveURL(
+      '?path=/docs/getting-started-welcome--welcome'
+    );
+  });
+
+  test('accessibility-checker keyboard navigation - item without href prop', async ({
```
Comment (tay1orjones) [inline e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js]:
```suggestion
  test('@avt-keyboard-nav - item without href prop', async ({
```

File: e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    await page.keyboard.press('Tab');
+    await page.keyboard.press('Tab');
+    await expect(page.getByText('Breadcrumb 3')).toBeFocused();
+
+    // The Breadcrumb 4 should not be focusable
+    await page.keyboard.press('Tab');
+    await expect(page.getByText('Breadcrumb 4')).not.toBeFocused();
+  });
+
+  test('accessibility-checker with overflow menu keyboard navigation', async ({
```
Comment (tay1orjones) [inline e2e/components/breadcrumb/breadcrumb-test.avt.e2e.js]:
```suggestion
  test('@avt-keyboard-nav with overflow menu keyboard navigation', async ({
```

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Small naming updates to help with https://github.com/carbon-design-system/carbon/issues/14595

**Review by tay1orjones — APPROVED** [review]
LGTM 👍

**Review by tw15egan — APPROVED** [review]

**Comment by guidari** [comment]
@tay1orjones FYI, 

I had to changed the way I was expecting this test to work to pass.  For some reason after clicking in a link, the playwright was failing in the CI and even with me increasing the timeout from default 3000ms to 120000ms, 500000ms… kept failing. 

While I was trying to figure out this issue, I found that the tests written with `test.slow()` are not being run in the CI neither locally. I found that once I download the report from playwiright and took a deeper look, the tests are not being addressed there. Maybe we might have to re-write those tests in another way?

Screenshot 2023-09-14 at 16 51 20

If we search the title of the test with `test.slow()` it doesn't appear in the `report.json` from playwright.
Also it is the screenshot with a long timeout there.

**Pull request merged**